app/controllers/cloud_volume_snapshot_controller.rb

```diff
@@ -36,15 +36,7 @@ def specific_buttons(pressed)
   def delete_cloud_volume_snapshots
     assert_privileges("cloud_volume_snapshot_delete")
 
-    snapshots = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "cloud_volume_snapshot")
-                  find_checked_ids_with_rbac(CloudVolumeSnapshot)
-                else
-                  [find_id_with_rbac(CloudVolumeSnapshot, params[:id])]
-                end
-
-    if snapshots.empty?
-      add_flash(_("No Cloud Volume Snapshots were selected for deletion."), :error)
-    end
+    snapshots = find_records_with_rbac(CloudVolumeSnapshot,checked_or_params)
```

Comment:
This change is not correct. In the code later you're expecting that `snapshots` contains IDs of the records, however `find_records_with_rbac` returns records.

The changes would be correct, if you'd assign the IDs of the records (`find_records_with_rbac(CloudVolumeSnapshot,checked_or_params).ids`).

But there is even better way:

You can see that the code works with IDs, but than it has to load the records from the database (again) on line 43:
`snapshot = CloudVolumeSnapshot.find_by_id(snapshot_id)`

Please, take a look at the changes @webe745 did in the commit https://github.com/ManageIQ/manageiq-ui-classic/pull/3796/commits/40b2e3d21c51d0965844787905c9adfa2933532f, and do the same here